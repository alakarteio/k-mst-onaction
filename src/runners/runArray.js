export default (dispatch, action, tree) => {
  dispatch.forEach(runner => runner(action, tree))
}
