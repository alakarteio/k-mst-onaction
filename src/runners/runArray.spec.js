/* eslint-env jest */
import runArray from './runArray'


describe('array runner', () => {
  it('should run all function from the array', () => {
    const first = jest.fn(() => 'first')
    const second = jest.fn(() => 'second')

    runArray([first, second], { ho: 'an action' }, { this: 'is tree !' })

    expect(first.mock.calls).toMatchSnapshot()
    expect(second.mock.calls).toMatchSnapshot()
  })

  it('should works with empty array', () => {
    runArray([], { ho: 'an action' }, { this: 'is tree !' })
  })
})
