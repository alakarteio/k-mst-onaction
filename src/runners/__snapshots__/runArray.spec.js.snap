// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array runner should run all function from the array 1`] = `
Array [
  Array [
    Object {
      "ho": "an action",
    },
    Object {
      "this": "is tree !",
    },
  ],
]
`;

exports[`array runner should run all function from the array 2`] = `
Array [
  Array [
    Object {
      "ho": "an action",
    },
    Object {
      "this": "is tree !",
    },
  ],
]
`;
